src/online/index.js

```diff
@@ -1,295 +1,267 @@
+      return () => {
+        let promise = service.retrieveMultipleRecords(infos)
+        putRecord口.push(multiPutBuilder()
+          .with(infos, id, () => loadingStateTransition())
+          .build()
+        )
+        promise.then(
+          records => {
+            putRecord口.push(multiPutBuilder()
+              .with(infos,   id, () => INITIAL_OPERATION_STATE)
+              .with(records, id, completedStateTransition)
+              .build()
+            )
+          },
+          error => {
+            putRecord口.push(multiPutBuilder()
+              .with(infos,   id, () => errorStateTransition(error))
+              .build()
+            )
+          },
+        )
+        .done()
       }
-    } else {
-      let error = new Error('Unauthenticated!')
-      error.isUnauthenticated = true
-      return Promise.reject(error)
     }
   }
 
   function Ranking(data) {
 
-    const descriptor   = new Descriptor(data)
-    const resubmit口   = new Bacon.Bus()
-    const reload口     = new Bacon.Bus()
-
-    const submission川 = (
-      reloadable川(
-        () => submitOrRetrieveRecord(data),
-        (
-          Bacon.once()
-          .merge(resubmit口)
-          .merge(user川.changes().filter(user => !!user).first())
-        )
+    const resubmit口 = new Bacon.Bus()
+    const reload口   = new Bacon.Bus()
+
+    return construct()
+
+    function construct() {
+
+      let record     = getRecordPart()
+      let scoreboard = getScoreboardPart(getScoreboardIntents(record))
+
+      let state川 = (
+        Bacon.combineTemplate({
+          submission: record.state川,
+          scoreboard: scoreboard.state川,
+        })
+        .map(conformState)
       )
-      .map(state => {
-        if (state.status === 'completed') {
-          return {
-            status: 'completed',
-            error:  null,
-            record: state.value.data,
-            rank:   state.value.meta.rank,
-          }
-        } else if (state.status === 'error' && state.error.isUnauthenticated) {
-          return {
-            status: 'unauthenticated',
-            error:  null,
-            record: null,
-            rank:   null,
+
+      return {
+        state川,
+        resubmit:         () => resubmit口.push(),
+        reloadScoreboard: () => reload口.push(),
+      }
+
+    }
+
+    // Make the state conform the old API. We should remove this in the future.
+    function conformState(state) {
+      return {
+        data: state.scoreboard.value && state.scoreboard.value.data,
+        meta: {
+          scoreboard: _.omit(state.scoreboard, 'value'),
+          submission: state.submission,
+        },
+      }
+    }
+
+    function getRecordPart() {
+
+      let operation川            = (
+        // We need to have it trigger on `resubmit口`, but cannot simply use
+        // `Bacon.when`, because of a weird issue: https://github.com/baconjs/bacon.js/issues/640
+        user川.combine(resubmit口.toProperty(null), user => user)
```

Comment:
This logic for deciding whether to submit or not should go to the composing function… (`construct`)